mempool/src/pool/tests.rs

```diff
@@ -0,0 +1,1787 @@
+        flags,
+        locktime,
+    )?;
+    mempool.add_transaction(child_tx)?;
+
+    let relay_fee = get_relay_fee_from_tx_size(TX_SPEND_INPUT_SIZE);
+    let replacement_fee = Amount::from_atoms(relay_fee + 100);
+    let replacement_tx =
+        tx_spend_input(&mempool, child_tx_input, replacement_fee, flags, locktime)?;
+    mempool.add_transaction(replacement_tx)?;
+    Ok(())
+}
+
+// To test our validation of BIP125 Rule#4 (replacement transaction pays for its own bandwidth), we need to know the necessary relay fee before creating the transaction. The relay fee depends on the size of the transaction. The usual way to get the size of a transaction is to call `tx.encoded_size` but we cannot do this until we have created the transaction itself. To get around this cycle, we have precomputed the size of all transaction created by `tx_spend_input`. This value will be the same for all transactions created by this function.
+const TX_SPEND_INPUT_SIZE: usize = 213;
+
+fn tx_spend_input<T: GetTime, M: GetMemoryUsage>(
+    mempool: &MempoolImpl<ChainStateMock, T, M>,
+    input: TxInput,
+    fee: impl Into<Option<Amount>>,
+    flags: u32,
+    locktime: u32,
+) -> anyhow::Result<Transaction> {
+    let fee = fee.into().map_or_else(
+        || Amount::from_atoms(get_relay_fee_from_tx_size(estimate_tx_size(1, 2))),
+        std::convert::identity,
+    );
+    tx_spend_several_inputs(mempool, &[input], fee, flags, locktime)
+}
+
+fn tx_spend_several_inputs<T: GetTime, M: GetMemoryUsage>(
+    mempool: &MempoolImpl<ChainStateMock, T, M>,
+    inputs: &[TxInput],
+    fee: Amount,
+    flags: u32,
+    locktime: u32,
+) -> anyhow::Result<Transaction> {
+    let input_value = inputs
+        .iter()
+        .map(|input| mempool.get_input_value(input))
+        .collect::<Result<Vec<_>, _>>()?
+        .into_iter()
+        .sum::<Option<_>>()
+        .ok_or_else(|| {
+            let msg = String::from("tx_spend_input: overflow");
+            log::error!("{}", msg);
+            anyhow::Error::msg(msg)
+        })?;
+
+    let available_for_spending = (input_value - fee).ok_or_else(|| {
+        let msg = format!(
+            "tx_spend_several_inputs: input_value ({:?}) lower than fee ({:?})",
+            input_value, fee
+        );
+        log::error!("{}", msg);
+        anyhow::Error::msg(msg)
+    })?;
+    let spent = (available_for_spending / 2).expect("division error");
+
+    let change = (available_for_spending - spent).ok_or_else(|| {
+        let msg = String::from("Error computing change");
+        anyhow::Error::msg(msg)
+    })?;
+
+    Transaction::new(
+        flags,
+        inputs.to_owned(),
+        vec![
+            TxOutput::new(spent, OutputPurpose::Transfer(Destination::AnyoneCanSpend)),
+            TxOutput::new(change, OutputPurpose::Transfer(Destination::AnyoneCanSpend)),
+        ],
+        locktime,
+    )
+    .map_err(Into::into)
+}
+
+#[test]
+fn one_ancestor_signal_is_enough() -> anyhow::Result<()> {
```

Comment:
What does signal mean here? Did you want to say "signature"?

Reply:
By signal here I meant an ancestor which has the replaceability flag set. The jargon is taken from the bip... Maybe I should be more explicit and rename this to "replaceability signal".

Reply:
Please call it that.

Reply:
Done.